Possibly incorrect paths in Turbinia config

This might very well be a lack of understanding on my end, but I tried uploading a file to Turbinia ([gpt.raw](https://github.com/google/turbinia/blob/9644e0519efc46c4456f1a5beb58f6cd3eb9210a/test_data/gpt.raw) from Turbinia test-data), with turbinia-client which failed with HTTP status 500.
```
turbinia-client evidence upload -j -p ./gpt.raw 123       
2024-03-26 15:28:09,192 INFO turbinia - Using configuration instance name -> default with host http://localhost:8000
2024-03-26 15:28:10,054 ERROR turbinia - Received status code 500 when calling upload_evidence: (500)
Reason: Internal Server Error
HTTP response headers: HTTPHeaderDict({'date': 'Tue, 26 Mar 2024 14:28:09 GMT', 'server': 'uvicorn', 'content-length': '21', 'content-type': 'text/plain; charset=utf-8'})
HTTP response body: Internal Server Error

{
  "/Users/user/Downloads/gpt.raw": "Received status code 500 when calling upload_evidence: (500)\nReason: Internal Server Error\nHTTP response headers: HTTPHeaderDict({'date': 'Tue, 26 Mar 2024 14:28:09 GMT', 'server': 'uvicorn', 'content-length': '21', 'content-type': 'text/plain; charset=utf-8'})\nHTTP response body: Internal Server Error\n"
}
```
Looking at the server logs the exception was:
```
2024-03-26 14:28:10 [ERROR] Failed in setting path for file gpt in ticket 123
INFO:     127.0.0.1:52302 - "POST /api/evidence/upload HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/turbinia/.venv/lib/python3.10/site-packages/uvicorn/protocols/http/httptools_impl.py", line 412, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/home/turbinia/.venv/lib/python3.10/site-packages/uvicorn/middleware/proxy_headers.py", line 69, in __call__
    return await self.app(scope, receive, send)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/fastapi/applications.py", line 284, in __call__
    await super().__call__(scope, receive, send)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/home/turbinia/.venv/lib/python3.10/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/starlette/middleware/cors.py", line 83, in __call__
    await self.app(scope, receive, send)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/home/turbinia/.venv/lib/python3.10/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/home/turbinia/.venv/lib/python3.10/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/home/turbinia/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 241, in app
    raw_response = await run_endpoint_function(
  File "/home/turbinia/.venv/lib/python3.10/site-packages/fastapi/routing.py", line 167, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/turbinia/turbinia/api/routes/evidence.py", line 260, in upload_evidence
    os.remove(file_path)
UnboundLocalError: local variable 'file_path' referenced before assignment
```
From a quick look at the `api/routes/evidence.py` it seems to reference a config variable called API_EVIDENCE_UPLOAD_DIR, which in my config is set to `/evidence`. This isn't a directory as far as I can see, neither is it mounted. It seems to be the default from the config downloaded in init-configmap

It also seems that the init-configmap for Turbinia sets some values to the PVC mount in the helm chart, but for some reason MOUNT_DIR_PREFIX seems to point to another directory, which isn't the PVC mount.
https://github.com/google/osdfir-infrastructure/blob/42b1774552d53091f8ae79f004037c356bd5eda6/charts/turbinia/templates/init-configmap.yaml#L56

Additional info: the stack is K3S with Longhorn, deployed with ArgoCD. I access turbinia through port-forwarding.

--- a/charts/turbinia/Chart.yaml
+++ b/charts/turbinia/Chart.yaml
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: turbinia
-version: 1.0.3
+version: 1.0.4
 description: A Helm chart for Turbinia Kubernetes deployments.
 keywords:
 - turbinia

--- a/charts/turbinia/templates/_initContainer.tpl
+++ b/charts/turbinia/templates/_initContainer.tpl
@@ -5,7 +5,7 @@ and Worker pod upon startup.
 */}}
 {{- define "turbinia.initContainer" -}}
 - name: init-turbinia
-  image: busybox
+  image: alpine
   command: ['sh', '-c', '/init/init-turbinia.sh']
   env:
     {{- if and .Values.redis.enabled .Values.redis.auth.enabled }}

--- a/charts/turbinia/templates/init-configmap.yaml
+++ b/charts/turbinia/templates/init-configmap.yaml
@@ -9,6 +9,7 @@ data:
   init-turbinia.sh: |
     #!/bin/sh
     set -e
+    apk add jq
 
     # Fix Filestore permissions
     chmod go+w /mnt/turbiniavolume
@@ -20,12 +21,18 @@ data:
     if [  $(ls /tmp/turbinia/ | wc -l) -gt 0 ]; then
       echo "Using exisiting configuration file provided."
       cp /tmp/turbinia/* /etc/turbinia/
-    else 
+    else
       # Pull default config if one is not already provided
-      echo -n "* Fetching configuration files.." 
-      # Fetch default Turbinia config file
+      {{- $turbiniaWorker := "us-docker.pkg.dev/osdfir-registry/turbinia/release/turbinia-worker" -}}
+      {{- if eq .Values.worker.image.repository $turbiniaWorker }}
+      echo -n "* Fetching the release Turbinia configuration file..." 
+      RELEASE_TAG=$(wget -O- https://api.github.com/repos/google/turbinia/releases | jq -r '.[0] | .tag_name')
+      wget "https://raw.githubusercontent.com/google/turbinia/$RELEASE_TAG/turbinia/config/turbinia_config_tmpl.py" -O turbinia.conf
+      {{- else }}
+      echo -n "* Fetching the latest Turbinia configuration file..." 
       wget "https://raw.githubusercontent.com/google/turbinia/master/turbinia/config/turbinia_config_tmpl.py" -O turbinia.conf
       echo "OK"
+      {{- end }}
     fi
 
     # Set up the Redis connection
@@ -55,6 +62,7 @@ data:
     sed -i -e "s/^LOG_DIR = .*$/LOG_DIR = '\/mnt\/{{ .Values.persistence.name }}\/logs'/g" turbinia.conf
     sed -i -e "s/^MOUNT_DIR_PREFIX = .*$/MOUNT_DIR_PREFIX = '\/mnt\/turbinia'/g" turbinia.conf
     sed -i -e "s/^OUTPUT_DIR = .*$/OUTPUT_DIR = '\/mnt\/{{ .Values.persistence.name }}\/output'/g" turbinia.conf
+    sed -i -e "s/^API_EVIDENCE_UPLOAD_DIR = .*$/API_EVIDENCE_UPLOAD_DIR = '\/mnt\/{{ .Values.persistence.name }}\/upload'/g" turbinia.conf
     
     # Disable Turbinia Jobs
     sed -i -e "s/^DISABLED_JOBS = .*$/DISABLED_JOBS = {{ .Values.config.disabledJobs }}/g" turbinia.conf

